src/context/graphql/extensions.test.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @jest-environment jsdom
+ */
+import { errors } from './errors'
+import { data } from './data'
+import { extensions } from './extensions'
+import { response } from '../../response'
+
+test('sets a given extensions on the response JSON body in the presence of no errors or data', async () => {
+  const result = await response(extensions({ tracking: { version: 1 } }))
+
+  expect(result.headers.get('content-type')).toEqual('application/json')
+  expect(result).toHaveProperty(
+    'body',
+    JSON.stringify({
+      extensions: {
+        tracking: {
+          version: 1,
+        },
+      },
+    }),
+  )
+})
+
+test('sets given extensions on the response JSON body in the presence of data', async () => {
+  const result = await response(
+    data({ hello: 'world' }),
+    extensions({ tracking: { version: 1 } }),
+  )
+
+  expect(result.headers.get('content-type')).toEqual('application/json')
+  expect(result).toHaveProperty(
+    'body',
+    JSON.stringify({
+      extensions: {
+        tracking: {
+          version: 1,
+        },
+      },
+      data: {
+        hello: 'world',
+      },
+    }),
+  )
+})
+
+test('sets given extensions on the response JSON body in the presence of data and errors', async () => {
```

Comment:
Thank you for adding these tests!

We'd also have to include a few integration tests (for browser and Node.js) so we're sure `ctx.extensions` behaves as expected in those environments when using a complete MSW setup. 

Would you be interested in adding such tests? 

It's a matter of adding two new test suites:

- `test/graphql-api/extensions.test.ts` (to test the feature's behavior in a browser)
- `test/graphql-api/extensions.node.test.ts` (to test the feature's behavior in Node.js)

You can take a look at the existing [browser](https://github.com/mswjs/msw/blob/master/test/graphql-api/query.test.ts) and [Node.js](https://github.com/mswjs/msw/blob/8bfcf6e39392790bb164c2dbd80cb28e8b57b140/test/graphql-api/cookies.node.test.ts#L50-L68) GraphQL integration tests for inspiration. 

> Note that you don't have to add an actual `createServer` with the GraphQL resolution. `setupServer` + GraphQL request handler using `ctx.extensions()` would be enough.

Reply:
Will add these today!

Reply:
Thank you for adding the tests!